doc/tikv/transaction-handling-newbie-perspective.md

```diff
@@ -0,0 +1,345 @@
+
+And then we'll use grpc call [`GetStore`](https://github.com/pingcap/kvproto/blob/d4aeb467de2904c19a20a12de47c25213b759da1/proto/pdpb.proto#L31) in `map_region_to_store` to find out the major replica of region.
+
+The remote fuction `GetStore` it defined is [here](https://github.com/pingcap/pd/blob/2b56a4c5915cb4b8806629193fd943a2e860ae4f/server/grpc_service.go#L171) in pd.
+
+Finally we'll get a `KvRpcClient` instance, which represents the connection to a TiKV replica.
+
+Then let's back to `retry_response_stream`, next function call we should pay attention to is  `store.dispatch`, it calls grpc function [`KvGet`](https://github.com/pingcap/kvproto/blob/d4aeb467de2904c19a20a12de47c25213b759da1/proto/TiKVpb.proto#L21).
+
+And finally we reach the code in TiKV's repo. In TiKV, the requests are handled by [`Server` struct](https://github.com/TiKV/TiKV/blob/e3058403a0fc9a96870882bf184ac075223b4642/src/server/server.rs#L48) , and the `KvGet` will be handled by `future_get` [here](https://github.com/TiKV/TiKV/blob/1de029631e09a3f9989a468a9cb4b97ec4db440e/src/server/service/kv.rs#L1155).
+
+Firstly we'll read the value for a key by using [`Storage::get`](https://github.com/TiKV/TiKV/blob/1de029631e09a3f9989a468a9cb4b97ec4db440e/src/storage/mod.rs#L216).
+
+`get` function is a little bit long, we'll ignore `STATIC` parts for now, and we'll get:
+
+```rust, no_run
+pub fn get(&self, mut ctx: Context, key: Key,
+    start_ts: TimeStamp) -> impl Future<Item = Option<Value>, Error = Error> {
+    const CMD: CommandKind = CommandKind::get;
+    let priority = ctx.get_priority();
+    let priority_tag = get_priority_tag(priority);
+
+    let res = self.read_pool.spawn_handle(
+        async move {
+            // The bypass_locks set will be checked at most once. `TsSet::vec` is more efficient
+            // here.
+            let bypass_locks = TsSet::vec_from_u64s(ctx.take_resolved_locks());
+            let snapshot = Self::with_tls_engine(|engine| Self::snapshot(engine, &ctx)).await?;
+            let snap_store = SnapshotStore::new(snapshot, start_ts,
+                        ctx.get_isolation_level(),
+                        !ctx.get_not_fill_cache(),
+                        bypass_locks,
+                        false);
+            let result = snap_store.get(&key, &mut statistics)
+                    // map storage::txn::Error -> storage::Error
+                    .map_err(Error::from);
+            result
+        },
+        priority,
+        thread_rng().next_u64(),
+    );
+    res.map_err(|_| Error::from(ErrorInner::SchedTooBusy))
+        .flatten()
+}
+```
+
+This function will get a `snapshot`, and then construct a `SnapshotStore` by using the `snapshot`, and then call `get` on this `SnapshotStore`, and finally get the data we need.
+
+(I don't really understand the `bypass_locks` part.)
```

Comment:
Can someone explain this to me?

Reply:
PTAL @MyonKeminta

Reply:
> Can someone give me some explaination about this?

I can understand this after read something about [tikv#5798](https://github.com/tikv/tikv/pull/5798) and [blog post about large transaction](https://pingcap.com/blog/large-transactions-in-tidb/).